node/src/blockchain/blockchain_interface_initializer.rs

```diff
@@ -46,21 +46,45 @@ mod tests {
     use crate::blockchain::blockchain_interface_initializer::BlockchainInterfaceInitializer;
     use masq_lib::blockchains::chains::Chain;
 
+    use futures::Future;
     use std::net::Ipv4Addr;
+    use web3::transports::Http;
 
-    use crate::blockchain::blockchain_interface::test_utils::test_blockchain_interface_is_connected_and_functioning;
-
+    use crate::blockchain::blockchain_interface::blockchain_interface_web3::{
+        BlockchainInterfaceWeb3, REQUESTS_IN_PARALLEL,
+    };
+    use crate::blockchain::blockchain_interface::BlockchainInterface;
+    use crate::test_utils::make_wallet;
     use masq_lib::constants::DEFAULT_CHAIN;
+    use masq_lib::test_utils::mock_blockchain_client_server::MBCSBuilder;
+    use masq_lib::utils::find_free_port;
 
     #[test]
     fn initialize_web3_interface_works() {
-        let subject_factory = |port: u16, chain: Chain| {
-            let subject = BlockchainInterfaceInitializer {};
-            let server_url = &format!("http://{}:{}", &Ipv4Addr::LOCALHOST.to_string(), port);
-            subject.initialize_web3_interface(server_url, chain)
-        };
+        let port = find_free_port();
+        let _blockchain_client_server = MBCSBuilder::new(port)
+            .response("0x3B9ACA00".to_string(), 0)
+            .response("0xFF40".to_string(), 0)
+            .response(
+                "0x000000000000000000000000000000000000000000000000000000000000FFFF".to_string(),
+                0,
+            )
+            .response("0x23".to_string(), 1)
+            .start();
+        let wallet = make_wallet("123");
+        let chain = Chain::PolyMainnet;
+        let server_url = &format!("http://{}:{}", &Ipv4Addr::LOCALHOST, port);
+        let (event_loop_handle, transport) =
+            Http::with_max_parallel(server_url, REQUESTS_IN_PARALLEL).unwrap();
+        let subject = BlockchainInterfaceWeb3::new(transport, event_loop_handle, chain);
+
+        let blockchain_agent = subject
+            .build_blockchain_agent(wallet.clone())
+            .wait()
+            .unwrap();
 
-        test_blockchain_interface_is_connected_and_functioning(subject_factory)
+        assert_eq!(blockchain_agent.consuming_wallet(), &wallet);
+        assert_eq!(blockchain_agent.agreed_fee_per_computation_unit(), 2);
```

Comment:
I don't mind this big facelift of this test but there are two things that need still be done for completeness.

1. 
I can see that the test proves the subject being able to "hear" what the blockchain_client_server tells to it.
I'd like to see you also assert on the communication working bidirectionally. On at least some of the requests     sent from the subject's end to the server. Theoretically, the server could produce the responses despite you sent the subject (or anything different than the subject) could've sent any kind of garbage, maybe "I don't believe that greenhouse gases have an effect", and the test wouldn't blow up.  (You should be able to use a method that belongs to the mock server to query the requests recorded inside it during the interaction.)

2.
I figured out you are asserting on a value that originated in the blockchain server and so it shows that the `agreed_fee_per_computation_unit` which is only a fancy name for "gas price" was made out during the build_blockchain_agent() call. It is what I'd expect that it happened.

The problem is the value 2 in the assertion. Because it should be a value read somehow from one of those responses hang up on the blockchain_client_server by its builder pattern. 
  
However, I cannot really read out which response it was that brought in this value from there. None of them contains the number two alone. 

So you maybe first should check this is really how the value got into the assertion. If you identify which response it is, please add a comment there, marking it.